src/main/java/reposense/parser/CsvParser.java

```diff
@@ -0,0 +1,158 @@
+     *
+     * Parameters - organization, repositoryName and branch are used to create the key to access repositoryMap.
+     */
+    private void createRepoConfigInMapIfNotExists(final HashMap<String, RepoConfiguration> repositoryMap,
+            final String organization, final String repositoryName, final String branch, final Date sinceDate,
+            final Date untilDate) {
+        final String key = createRepoConfigKey(organization, repositoryName, branch);
+
+        if (!repositoryMap.containsKey(key)) {
+            RepoConfiguration config = new RepoConfiguration(organization, repositoryName, branch);
+
+            config.setSinceDate(sinceDate);
+            config.setUntilDate(untilDate);
+
+            repositoryMap.put(key, config);
+        }
+    }
+
+    /**
+     * Returns a {@code RepoConfiguration} contained in {@code repositoryMap}
+     *
+     * Parameters - organization, repositoryName and branch are used to create the key to access repositoryMap.
+     */
+    private RepoConfiguration getRepoConfigFromMap(final HashMap<String, RepoConfiguration> repositoryMap,
+            final String organization, final String repositoryName, final String branch,
+            final Date sinceDate, final Date untilDate) {
+        final String key = createRepoConfigKey(organization, repositoryName, branch);
+        createRepoConfigInMapIfNotExists(repositoryMap, organization, repositoryName, branch, sinceDate, untilDate);
+
+        return repositoryMap.get(key);
+    }
+
+    private String createRepoConfigKey(String organization, String repositoryName, String branch) {
+        return String.format(REPO_CONFIG_MAP_KEY_FORMAT, organization, repositoryName, branch);
+    }
+
+    /**
+     * Returns a list of {@code RepoConfiguration}, which are the inflated object a line of the csv file.
+     *
+     * @throws IOException If user-supplied csv file does not exists or is not readable.
+     */
+    public List<RepoConfiguration> parse(CliArguments argument) throws IOException {
+        assert (argument != null);
+
+        final Date sinceDate = argument.getSinceDate().orElse(null);
+        final Date untilDate = argument.getUntilDate().orElse(null);
+        final Path path = argument.getConfigFile().toPath();
+
+        HashMap<String, RepoConfiguration> repositoryMap = new HashMap<String, RepoConfiguration>();
+        int lineNumber = 1;
+
+        try {
+            // Skips first line, which is the header row
+            final Collection<String> lines = Files.lines(path).skip(SKIP_FIRST_LINE).collect(Collectors.toList());
+
+            for (final String line : lines) {
+                processLine(repositoryMap, sinceDate, untilDate, line, lineNumber);
+                lineNumber++;
+            }
+
+        } catch (IOException iex) {
+            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE);
+        }
+
+        return new ArrayList<RepoConfiguration>(repositoryMap.values());
```

Comment:
the levels of abstraction in this method is quite uneven.

Also, can you try this instead?

If repoconfig exists -> add author
else -> create and add the repoconfig to the list.